File: 05_checkSparseMatrix/5.c
#include <stdio.h>
#define MAX 100

int isSparseMatrix(int array[][MAX], int m, int n) {
    int count = 0;

    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            if (array[i][j] == 0) {
                count++;
            }
        }
    }

    return (count > (m * n) / 2);
}

int main() {
    int array[3][MAX] = {
        {1, 0, 3},
        {0, 0, 4},
        {6, 0, 0}
    };
    int m = 3, n = 3;

    if (isSparseMatrix(array, m, n)) {
        printf("Yes\n");
    } else {
        printf("No\n");
    }

    return 0;
}

File: 05_checkSparseMatrix/5.py
MAX = 100
 
def isSparseMatrix(array,m, n) :
    
    count = 0

    for i in range(0,m) :
        for j in range(0,n) :
            if (array[i][j] == 0) :
                count = count + 1
 
    return (count > 
            ((m * n) // 2))

array = [ [ 1, 0, 3 ],
          [ 0, 0, 4 ],
          [ 6, 0, 0 ] ]
m = 3
n = 3

if (isSparseMatrix(array, m, n)) :
    print("Yes")
else :
    print("No")
